src/module.c

```diff
@@ -1964,6 +1958,12 @@ int RM_GetContextFlags(RedisModuleCtx *ctx) {
         }
     }
 
+    if (server.in_eval)
+        flags |= REDISMODULE_CTX_FLAGS_LUA;
```

Comment:
i just realized this can cause the wrong indication in case of a busy script.
however, in theory, module commands can never run during busy scripts (the list of commands that are allowed while server.lua_timedout is hard coded in processCommand, and AFAIK, we never release the GIL during these).

maybe @MeirShpilraien should take a look just in case.